syntaxes/src/expression/grammar.ts

```diff
@@ -0,0 +1,886 @@
+import { GrammarDefinition } from '../types';
+
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export const expression: GrammarDefinition = {
+  scopeName: 'expression.ng',
+  injectionSelector: 'L:text.html -comment',
+  patterns: [{ include: '#ngExpression' }],
+  repository: {
+    ngExpression: {
+      name: 'meta.expression.ng',
+      patterns: [
+        {
+          include: '#tsString',
+        },
+        {
+          include: '#literal',
+        },
+        {
+          include: '#ternaryExpression',
+        },
+        {
+          include: '#expressionOperators',
+        },
+        {
+          include: '#functionCall',
+        },
+        {
+          include: '#identifiers',
+        },
+        {
+          include: '#parenExpression',
+        },
+        {
+          include: '#punctuationComma',
+        },
+        {
+          include: '#punctuationAccessor',
+        },
+      ],
+    },
+
+    arrayLiteral: {
+      name: 'meta.array.literal.ts',
+      begin: /\[/,
+      beginCaptures: {
+        '0': {
+          name: 'meta.brace.square.ts',
+        },
+      },
+      end: /\]/,
+      endCaptures: {
+        '0': {
+          name: 'meta.brace.square.ts',
+        },
+      },
+      patterns: [
+        {
+          include: '#ngExpression',
+        },
+        {
+          include: '#punctuationComma',
+        },
+      ],
+    },
+
+    booleanLiteral: {
+      patterns: [
+        {
+          name: 'constant.language.boolean.true.ts',
+          match: /(?<!\.|\$)\btrue\b(?!\$)/,
+        },
+        {
+          name: 'constant.language.boolean.false.ts',
+          match: /(?<!\.|\$)\bfalse\b(?!\$)/,
+        },
+      ],
+    },
+
+    expressionOperators: {
+      patterns: [
+        {
+          match: /((?<!\|)(\|)(?!\|))\s?([a-zA-Z0-9\-\_]*)/,
+          captures: {
+            '2': {
+              name: 'keyword.operator.logical.ts',
+            },
+            '3': {
+              name: 'entity.name.function.pipe.ng',
+            },
+          },
+        },
+        {
+          name: 'storage.type.ts',
+          match: /(?<!\.|\$)\b(let)\b(?!\$)/,
```

Comment:
Similar to above, what is the purpose of matching a literal `$` via lookahead?

Reply:
I kept this in `let` to keep the pattern that was already being used in the base file. I believe the negative lookahead and lookbehind are basically preventing the scope `storage.type.ts` to being added to `.let`, `$let` and `let$`. A trailing `$` is often used as a variable naming convention for observables.

Reply:
Okay, I see. Wow, they really are covering all the cases..